middleware/service_now.go

```diff
@@ -101,27 +123,110 @@ func (s *ServiceNowService) ListSchemas(params map[string]interface{}) (interfac
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	return nil, fmt.Errorf("not implemented")
+	uri := os.Getenv("SERVICENOW_SCHEMA_LIST_API_PATH")
+	if uri == "" {
+		return nil, fmt.Errorf("failed to fetch business object models; SERVICENOW_SCHEMA_LIST_API_PATH not set")
+	}
+
+	status, resp, err := s.Get(uri, params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch business object models; status: %v; %s", status, err.Error())
+	}
+
+	if status != 200 {
+		return nil, fmt.Errorf("failed to fetch business object models; status: %v", status)
+	}
+
+	if arr, ok := resp.([]interface{}); ok {
+		schemas := make([]interface{}, 0)
+
+		for _, item := range arr {
+			var systemSchema map[string]interface{}
+			raw, _ := json.Marshal(item)
+
+			err = json.Unmarshal(raw, &systemSchema)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal; status: %v; %s", status, err.Error())
+			}
+
+			schemas = append(schemas, map[string]interface{}{
+				"description": systemSchema["table"],
+				"name":        systemSchema["table"],
+				"system_type": "servicenow",
+				"type":        systemSchema["id"],
+			})
+		}
+
+		return schemas, nil
+	}
+
+	return nil, fmt.Errorf("failed to fetch business object models; failed to parse response")
 }
 
 // GetSchema retrieves a business object model by type
 func (s *ServiceNowService) GetSchema(recordType string, params map[string]interface{}) (interface{}, error) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	return nil, fmt.Errorf("not implemented")
+	schemaDetailsPath := os.Getenv("SERVICENOW_SCHEMA_DETAILS_API_PATH")
+	if schemaDetailsPath == "" {
+		return nil, fmt.Errorf("failed to fetch business object model; SERVICENOW_SCHEMA_DETAILS_API_PATH not set")
+	}
+
+	uri := fmt.Sprintf("%s?table=%s", schemaDetailsPath, recordType)
+	status, resp, err := s.Get(uri, params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch business object model; status: %v; %s", status, err.Error())
+	}
+
+	if status != 200 {
+		return nil, fmt.Errorf("failed to fetch business object model; status: %v", status)
+	}
+
+	var _resp map[string]interface{}
+	raw, _ := json.Marshal(resp)
+
+	err = json.Unmarshal(raw, &_resp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch business object model; %s", err.Error())
+	}
+
+	if arr, ok := _resp["fields"].([]interface{}); ok {
+		fields := make([]interface{}, 0)
+
+		for _, item := range arr {
+			var _item map[string]interface{}
+			raw, _ := json.Marshal(item)
+
+			err = json.Unmarshal(raw, &_item)
+			if err != nil {
+				return nil, fmt.Errorf("failed to fetch business object model; %s", err.Error())
+			}
+
+			fields = append(fields, map[string]interface{}{
+				"name":        _item["display_value"],
+				"description": _item["internal_type"],
+				"type":        _item["internal_value"],
```

Comment:
`display_value`, `internal_type`, `internal_value`?

Reply:
@adunne09 are you sure about this mapping?

Reply:
yes absolutely